services/indexer/indexer.go

```diff
@@ -195,9 +195,9 @@ func (i *Indexer) mostRecentlySyncedBlockHeight(ctx context.Context) (cid.Cid, i
 	task, err := i.storage.MostRecentSyncedBlock(ctx)
 	if err != nil {
 		if err == pg.ErrNoRows {
-			return i.genesis.Cids()[0], 0, xerrors.Errorf("query recent synced: %w", err)
+			return i.genesis.Cids()[0], 0, nil
 		}
-		return cid.Undef, 0, err
+		return cid.Undef, 0, xerrors.Errorf("query recent synced: %w", err)
```

Comment:
I think the first return value is wrong.

Reply:
Can you explain more?

Reply:
Ah, I misunderstood what was happening here in my haste to review. GTM